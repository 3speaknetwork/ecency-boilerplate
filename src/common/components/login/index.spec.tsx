import React from "react";

import { Login } from "./index";
import renderer from "react-test-renderer";

const defProps = {
  users: [],
  activeUser: null,
  setActiveUser: () => {},
  updateActiveUser: () => {},
  deleteUser: () => {},
  onLogin: () => {},
};

it("(1) Default render", () => {
  const component = renderer.create(<Login {...defProps} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it("(2) With users", () => {
  const users = [
    {
      username: "user1",
      accessToken: "aa",
      refreshToken: "bb",
      expiresIn: 2,
    },
    {
      username: "user2",
      accessToken: "aa",
      refreshToken: "bb",
      expiresIn: 2,
    },
  ];

  const props = { ...defProps, users };

  const component = renderer.create(<Login {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it("(3) With users and active user", () => {
  const users = [
    {
      username: "user1",
      accessToken: "aa",
      refreshToken: "bb",
      expiresIn: 2,
    },
    {
      username: "user2",
      accessToken: "aa",
      refreshToken: "bb",
      expiresIn: 2,
    },
  ];

  const activeUser = {
    username: "user2",
    data: {
      name: "user2",
    },
  };

  const props = { ...defProps, users, activeUser };

  const component = renderer.create(<Login {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
