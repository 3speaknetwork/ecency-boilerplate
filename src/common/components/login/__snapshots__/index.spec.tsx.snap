// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
Array [
  <div
    className="dialog-header"
  >
    <img
      alt="Logo"
      src="logo-circle.svg"
    />
    <h2>
      Welcome back!
    </h2>
  </div>,
  <div
    className="login-form"
  >
    <p
      className="login-form-text"
    >
      Login with your username and private key.
    </p>
    <div
      className="form-group"
    >
      <input
        autoFocus={true}
        className="form-control"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="username"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="master or active private key"
        type="password"
        value=""
      />
    </div>
    <p
      className="login-form-text"
    >
      We don't store your private keys.
       
      <a
        href="#"
        onClick={[Function]}
      >
        More info
      </a>
    </p>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </div>,
  <div
    className="or-divider"
  >
    OR
  </div>,
  <div
    className="hs-login"
  >
    <a
      className="btn btn-outline-primary"
      href="https://hivesigner.com/oauth2/authorize?client_id=ecency.app&redirect_uri=http%3A%2F%2Flocalhost%2Fauth&response_type=code&scope=vote%2Ccomment%2Cdelete_comment%2Ccomment_options%2Ccustom_json%2Cclaim_reward_balance%2Coffline"
    >
      <img
        alt="hivesigner"
        className="hs-logo"
        src="hive-signer.svg"
      />
       
      Login with hivesigner
    </a>
  </div>,
  <p>
    Don't have an account?
     
    <a
      href="#"
      onClick={[Function]}
    >
      Sign up now!
    </a>
  </p>,
]
`;

exports[`(2) With users 1`] = `
Array [
  <div
    className="user-list"
  >
    <div
      className="user-list-header"
    >
      Login As
    </div>
    <div
      className="user-list-body"
    >
      <div
        className="user-list-item "
        onClick={[Function]}
      >
        <span
          className="user-avatar normal"
          style={
            Object {
              "backgroundImage": "url(https://images.ecency.com/u/user1/avatar/small)",
            }
          }
        />
        <span
          className="username"
        >
          @
          user1
        </span>
        <div
          className="flex-spacer"
        />
        <div
          className="btn-delete"
          onClick={[Function]}
        >
          <span
            title="Delete"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M6,19A2,2 0 0,0 8,21H16A2,2 0 0,0 18,19V7H6V19M8.46,11.88L9.87,10.47L12,12.59L14.12,10.47L15.53,11.88L13.41,14L15.53,16.12L14.12,17.53L12,15.41L9.88,17.53L8.47,16.12L10.59,14L8.46,11.88M15.5,4L14.5,3H9.5L8.5,4H5V6H19V4H15.5Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="user-list-item "
        onClick={[Function]}
      >
        <span
          className="user-avatar normal"
          style={
            Object {
              "backgroundImage": "url(https://images.ecency.com/u/user2/avatar/small)",
            }
          }
        />
        <span
          className="username"
        >
          @
          user2
        </span>
        <div
          className="flex-spacer"
        />
        <div
          className="btn-delete"
          onClick={[Function]}
        >
          <span
            title="Delete"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M6,19A2,2 0 0,0 8,21H16A2,2 0 0,0 18,19V7H6V19M8.46,11.88L9.87,10.47L12,12.59L14.12,10.47L15.53,11.88L13.41,14L15.53,16.12L14.12,17.53L12,15.41L9.88,17.53L8.47,16.12L10.59,14L8.46,11.88M15.5,4L14.5,3H9.5L8.5,4H5V6H19V4H15.5Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="or-divider"
  >
    OR
  </div>,
  <div
    className="login-form"
  >
    <p
      className="login-form-text"
    >
      Login with your username and private key.
    </p>
    <div
      className="form-group"
    >
      <input
        autoFocus={true}
        className="form-control"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="username"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="master or active private key"
        type="password"
        value=""
      />
    </div>
    <p
      className="login-form-text"
    >
      We don't store your private keys.
       
      <a
        href="#"
        onClick={[Function]}
      >
        More info
      </a>
    </p>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </div>,
  <div
    className="or-divider"
  >
    OR
  </div>,
  <div
    className="hs-login"
  >
    <a
      className="btn btn-outline-primary"
      href="https://hivesigner.com/oauth2/authorize?client_id=ecency.app&redirect_uri=http%3A%2F%2Flocalhost%2Fauth&response_type=code&scope=vote%2Ccomment%2Cdelete_comment%2Ccomment_options%2Ccustom_json%2Cclaim_reward_balance%2Coffline"
    >
      <img
        alt="hivesigner"
        className="hs-logo"
        src="hive-signer.svg"
      />
       
      Login with hivesigner
    </a>
  </div>,
  <p>
    Don't have an account?
     
    <a
      href="#"
      onClick={[Function]}
    >
      Sign up now!
    </a>
  </p>,
]
`;

exports[`(3) With users and active user 1`] = `
Array [
  <div
    className="user-list"
  >
    <div
      className="user-list-header"
    >
      Login As
    </div>
    <div
      className="user-list-body"
    >
      <div
        className="user-list-item "
        onClick={[Function]}
      >
        <span
          className="user-avatar normal"
          style={
            Object {
              "backgroundImage": "url(https://images.ecency.com/u/user1/avatar/small)",
            }
          }
        />
        <span
          className="username"
        >
          @
          user1
        </span>
        <div
          className="flex-spacer"
        />
        <div
          className="btn-delete"
          onClick={[Function]}
        >
          <span
            title="Delete"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M6,19A2,2 0 0,0 8,21H16A2,2 0 0,0 18,19V7H6V19M8.46,11.88L9.87,10.47L12,12.59L14.12,10.47L15.53,11.88L13.41,14L15.53,16.12L14.12,17.53L12,15.41L9.88,17.53L8.47,16.12L10.59,14L8.46,11.88M15.5,4L14.5,3H9.5L8.5,4H5V6H19V4H15.5Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="user-list-item active"
        onClick={[Function]}
      >
        <span
          className="user-avatar normal"
          style={
            Object {
              "backgroundImage": "url(https://images.ecency.com/u/user2/avatar/small)",
            }
          }
        />
        <span
          className="username"
        >
          @
          user2
        </span>
        <div
          className="check-mark"
        />
        <div
          className="flex-spacer"
        />
        <div
          className="btn-delete"
          onClick={[Function]}
        >
          <span
            title="Delete"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M6,19A2,2 0 0,0 8,21H16A2,2 0 0,0 18,19V7H6V19M8.46,11.88L9.87,10.47L12,12.59L14.12,10.47L15.53,11.88L13.41,14L15.53,16.12L14.12,17.53L12,15.41L9.88,17.53L8.47,16.12L10.59,14L8.46,11.88M15.5,4L14.5,3H9.5L8.5,4H5V6H19V4H15.5Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="or-divider"
  >
    OR
  </div>,
  <div
    className="login-form"
  >
    <p
      className="login-form-text"
    >
      Login with your username and private key.
    </p>
    <div
      className="form-group"
    >
      <input
        autoFocus={true}
        className="form-control"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="username"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="master or active private key"
        type="password"
        value=""
      />
    </div>
    <p
      className="login-form-text"
    >
      We don't store your private keys.
       
      <a
        href="#"
        onClick={[Function]}
      >
        More info
      </a>
    </p>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </div>,
  <div
    className="or-divider"
  >
    OR
  </div>,
  <div
    className="hs-login"
  >
    <a
      className="btn btn-outline-primary"
      href="https://hivesigner.com/oauth2/authorize?client_id=ecency.app&redirect_uri=http%3A%2F%2Flocalhost%2Fauth&response_type=code&scope=vote%2Ccomment%2Cdelete_comment%2Ccomment_options%2Ccustom_json%2Cclaim_reward_balance%2Coffline"
    >
      <img
        alt="hivesigner"
        className="hs-logo"
        src="hive-signer.svg"
      />
       
      Login with hivesigner
    </a>
  </div>,
]
`;
